Tried afew combinations of `nix` / `nixStatic` / `nixVersions.nix_2_28` and `pkgsStatic` / `pkgsCross.aarch64-multiplatform` / `pkgsCross.aarch64-multiplatform-musl`, and the only one that seems borked in my testing is `pkgsCross.aarch64-multiplatform-musl.nix` (after pulling in the fix for musl `which` from staging). But the failure is in `libexpr` and thus not a result of this PR.

```
FAILED: [code=1] libnixexpr.so 
aarch64-unknown-linux-musl-g++  -o libnixexpr.so libnixexpr.so.p/meson-generated_.._parser-tab.cc.o libnixexpr.so.p/meson-generated_.._lexer-tab.cc.o libnixexpr.so.p/attr-path.cc.o libnixexpr.so.p/attr-set.cc.o libnixexpr.so.p/eval-cache.cc.o libnixexpr.so.p/eval-error.cc.>
/nix/store/31fr4z3y0b5zsqagvxkwnvc7zqzjw3n7-fortify-headers-1.1alpine3/include/stdio.h: In function '__to_xstring.constprop':
/nix/store/31fr4z3y0b5zsqagvxkwnvc7zqzjw3n7-fortify-headers-1.1alpine3/include/stdio.h:80:28: error: inlining failed in call to 'always_inline' 'vsnprintf': function body can be overwritten at link time
   80 | _FORTIFY_FN(vsnprintf) int vsnprintf(char * _FORTIFY_POS0 __s, size_t __n,
      |                            ^
```

--- a/pkgs/tools/package-management/nix/common-meson.nix
+++ b/pkgs/tools/package-management/nix/common-meson.nix
@@ -59,10 +59,7 @@ assert (hash == null) -> (src != null);
   xz,
   enableDocumentation ? stdenv.buildPlatform.canExecute stdenv.hostPlatform,
   enableStatic ? stdenv.hostPlatform.isStatic,
-  withAWS ?
-    lib.meta.availableOn stdenv.hostPlatform aws-c-common
-    && !enableStatic
-    && (stdenv.hostPlatform.isLinux || stdenv.hostPlatform.isDarwin),
+  withAWS ? lib.meta.availableOn stdenv.hostPlatform aws-c-common,
   aws-c-common,
   aws-sdk-cpp,
   withLibseccomp ? lib.meta.availableOn stdenv.hostPlatform libseccomp,

--- a/pkgs/tools/package-management/nix/modular/src/libstore/package.nix
+++ b/pkgs/tools/package-management/nix/modular/src/libstore/package.nix
@@ -25,9 +25,7 @@
 
   withAWS ?
     # Default is this way because there have been issues building this dependency
-    stdenv.hostPlatform == stdenv.buildPlatform
-    && (stdenv.isLinux || stdenv.isDarwin)
-    && lib.meta.availableOn stdenv.hostPlatform aws-c-common,
+    lib.meta.availableOn stdenv.hostPlatform aws-c-common,
 }:
 
 mkMesonLibrary (finalAttrs: {

